### PR TITLE
Log analytic events for CC Libraries

### PR DESCRIPTION
**Pull request by shaoshing**
Here are the events logged:

* create/delete/update (only rename)/switch library
* create/delete/update (only rename) element
* apply color / color-theme / layerstyle / characterstyle
* place graphic
* open/update graphic
* open library links: share library, share public link, view online
* open element links: share public link, view online
* open Adobe Stock link

### CODE REVIEW

File: src/js/actions/libraries.js [hunk 1]
```diff
@@ -1199,6 +1200,36 @@ define(function (require, exports) {
...
+                    
+                    switch (event) {
+                        case "createLibrary":
+                            headlights.logEvent("libraries", "library", "create");
+                            break;
+                        case "deleteLibrary":
+                            headlights.logEvent("libraries", "library", "delete");
+                            break;
+                        case "createElement":
+                            //
```
Comment (iwehrman) [inline src/js/actions/libraries.js]: Missing helpful comment here? Or stray, stale comment?

File: src/js/util/headlights.js [hunk 2]
```diff
@@ -44,6 +45,7 @@ define(function (require, exports) {
      */
     var logEvent = function (category, subcategory, event) {
         if (global.debug) {
+            log.info("headlights: %s > %s > %s", category, subcategory, event);
```
Comment (iwehrman) [inline src/js/util/headlights.js]: I think this should be removed before this is merged. I know it only happens in debug mode only, but I'm still interested in keep the console output readable for developers, which means trying to limit the output to that which is widely useful. I think, outside of testing this PR, this probably won't be widely useful.

File: src/js/jsx/sections/libraries/LibraryList.jsx [hunk 3]
```diff
@@ -253,6 +254,20 @@ define(function (require, exports, module) {
             
             this.setState({ command: null });
         },
+        
+        /**
+         * Handle library button clicked
+         *
+         * @private
+         */
+        _handleLibraryButtonClicked: function (event, url) {
```
Comment (iwehrman) [inline src/js/jsx/sections/libraries/LibraryList.jsx]: Missing `@param` annotations.

File: src/js/jsx/sections/libraries/assets/AssetSection.jsx [hunk 4]
```diff
@@ -162,6 +161,20 @@ define(function (require, exports, module) {
         _handleEndEditingTitle: function () {
             this._isEditingTitle = false;
         },
+            
+        /**
+         * Handle element button clicked
+         *
+         * @private
+         */
+        _handleElementButtonClicked: function (event, url) {
```
Comment (iwehrman) [inline src/js/jsx/sections/libraries/assets/AssetSection.jsx]: Missing `@param` annotations.

File: src/js/jsx/sections/libraries/LibraryBar.jsx [hunk 5]
```diff
@@ -169,14 +169,28 @@ define(function (require, exports, module) {
...
         _handleSyncLibraries: function () {
             this.getFlux().actions.libraries.syncLibraries();
         },
+        
+        /**
+         * Handle stock button clicked
+         *
+         * @private
+         */
+        _handleStockButtonClicked: function () {
```
Comment (iwehrman) [inline src/js/jsx/sections/libraries/LibraryBar.jsx]: Missing `@param` annotations.

**Comment by iwehrman** [comment]
Fantastic! Just a few minor changes suggested. We'll get this merged before the standup. Back to @shaoshing.

**Comment by iwehrman** [comment]
Also, just want to confirm that you explicitly opted not to use `CCLibraries.analyticsUtils.reportUseElement` for a good reason. Like, e.g., it doesn't actually serve any purpose?

**Comment by shaoshing** [comment]
@iwehrman thanks for reviewing. I've addressed your comments. I actually missed the `reportUseElement` util function; but my first looks shows it is similar to using the `headlights.logEvent`, and require more codes to emit the events and then handle them together in the `reportEvent` callback, so probably not worth a lot to switch to it. Please let me know if you disagree.

**Comment by iwehrman** [comment]
Based on my limited understanding, I don't disagree with your decision to skip `reportUseElement`.

**Pull request merged**